Python/Product/PythonTools/PythonTools/LanguageServerClient/StreamHacking/StreamIntercepter.cs

```diff
@@ -44,10 +44,15 @@ class StreamIntercepter : Stream {
 
         public override void Flush() => baseStream.Flush();
         public override int Read(byte[] buffer, int offset, int count) {
-            var result = baseStream.Read(buffer, offset, count);
-            var args = new StreamData { bytes = buffer, offset = offset, count = result };
-            readHandler.Invoke(args);
-            return result;
+            var bytesRead = baseStream.Read(buffer, offset, count);
+            var args = new StreamData { bytes = buffer, offset = offset, count = bytesRead };
+            var newBytesRead = readHandler.Invoke(args);
+
+            if (newBytesRead != bytesRead) {
```

Comment:
I think this copy is redundant. You're already copying to the buffer passed in inside the read handler.